zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/SaslServerCallbackHandlerTest.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.zookeeper.server.auth;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.AppConfigurationEntry.LoginModuleControlFlag;
+import javax.security.auth.login.Configuration;
+import org.junit.Test;
+
+public class SaslServerCallbackHandlerTest {
+
+    @Test
+    public void wrongUserNameShouldClearUserCredential() throws IOException, UnsupportedCallbackException {
+        Configuration configuration = mock(Configuration.class);
+        Map<String, String> userCredentials = new HashMap<>();
+        userCredentials.put("user_exist_user", "password");
+        AppConfigurationEntry appConfigurationEntry = new AppConfigurationEntry("test-module", LoginModuleControlFlag.REQUIRED, userCredentials);
+        when(configuration.getAppConfigurationEntry("Server")).thenReturn(new AppConfigurationEntry[]{appConfigurationEntry});
```

Comment:
It would be nice to avoid mock. I saw multiple "Sasl*Test". I think they might be helpful.

Reply:
I'll give it a try.